pkg/networkservice/chains/endpoint/server.go

```diff
@@ -103,16 +103,16 @@ func NewServer(ctx context.Context, tokenGenerator token.GeneratorFunc, options
 	rv.NetworkServiceServer = chain.NewNamedNetworkServiceServer(
 		opts.name,
 		append([]networkservice.NetworkServiceServer{
+			serialize.NewServer(),
```

Comment:
`serialize` cannot stand before `updatepath` - in such case all Request/Close events generated from `refresh`/`timeout`/`heal` will race with Request/Close coming from previous hop, because they will have different `Connection.Id`.

Reply:
It seems as a problem with `serialize `.  But ok, I'll fix sequence because @edwarnicke planning to replace `serialize` to `begin`.

Reply:
@Bolodya1997 Do you have any idea why all tests working fine when `serialize` is first in the chain? Did we miss scenario coverage tests?

Reply:
CI for chain when serialize is first https://github.com/networkservicemesh/sdk/actions/runs/1040411831

Reply:
We don't have sandbox tests for timeout/refresh overlapping with Close/Request coming from the Client - it is pretty hard to guarantee such case without using `clock` in sandbox.